os/fs/tmpfs/fs_tmpfs.c

```diff
@@ -96,6 +96,27 @@
 #define tmpfs_unlock_directory(tdo) \
 	(tmpfs_unlock_object((FAR struct tmpfs_object_s *)tdo))
 
+#ifdef CONFIG_MM_NHEAPS
```

Comment:
I think `#ifdef CONFIG_MM_NHEAPS` is not needed.
Only `#if CONFIG_MM_NHEAPS > 1` is enough.